### PR TITLE
Resolved Task #5

### PR DESCRIPTION
**Pull request by wladengine**

### CODE REVIEW

File: MultiThreading.Task5.Threads.SharedCollection/Program.cs [hunk 1]
```diff
@@ -5,11 +5,21 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace MultiThreading.Task5.Threads.SharedCollection;
 
 class Program
 {
+    private const int CollectionLength = 10;
+    private static readonly ManualResetEventSlim _producerAddedNewValue = new(false);
```
Comment (ASemykinBrl) [inline MultiThreading.Task5.Threads.SharedCollection/Program.cs:17]: It seems like a single ManualResetEventSlim would be enough to synchronize 22 threads, isn't it?

File: MultiThreading.Task5.Threads.SharedCollection/Program.cs [hunk 2]
```diff
@@ -5,11 +5,21 @@
...
+    private static readonly ManualResetEventSlim _producerFinishedWork = new(false);
+    private static readonly ManualResetEventSlim _consumerHandledValue = new(false);
+    private static readonly List<int> _collection = new();
+    private static readonly object lockObject = new();
```
Comment (ASemykinBrl) [inline MultiThreading.Task5.Threads.SharedCollection/Program.cs]: Suppose we don't need lock at all? I removed it and nothing has changed

**Review by ASemykinBrl — CHANGES_REQUESTED** [review]
Some code cleanup is required

File: MultiThreading.Task5.Threads.SharedCollection/Program.cs (hunk 2 above)
Comment (wladengine) [reply MultiThreading.Task5.Threads.SharedCollection/Program.cs]: Yes, you are right, current synchronization mechanism isolates enough our collection.

**Review by ASemykinBrl — APPROVED** [review]

**Pull request merged**